ui/src/app/applications/components/application-details/application-details.tsx

```diff
@@ -459,14 +453,43 @@ export class ApplicationDetails extends React.Component<RouteComponentProps<{nam
         ];
     }
 
-    private filterTreeNode(node: ResourceTreeNode, filterInput: {kind: string[]; health: string[]; sync: string[]}): boolean {
+    private filterTreeNode(tree: ApplicationTree, node: ResourceTreeNode, filterInput: FilterInput, ownership?: string): boolean {
         const syncStatuses = filterInput.sync.map(item => (item === 'OutOfSync' ? ['OutOfSync', 'Unknown'] : [item])).reduce((first, second) => first.concat(second), []);
 
-        return (
+        const minutesAgo = (m: number) => {
+            const d = new Date();
+            d.setTime(d.getTime() - m * 60000);
+            return d;
+        };
+        const createdAt = new Date(node.createdAt); // will be falsely if the node has not been created, and so will not appear
+        const createdWithin = (n: number) => createdAt.getTime() > minutesAgo(n).getTime();
+        const labels = Object.entries(node.labels || {}).map(([name, value]) => name + '=' + value);
+
+        if (
             (filterInput.kind.length === 0 || filterInput.kind.indexOf(node.kind) > -1) &&
             (syncStatuses.length === 0 || node.root.hook || (node.root.status && syncStatuses.indexOf(node.root.status) > -1)) &&
-            (filterInput.health.length === 0 || node.root.hook || (node.root.health && filterInput.health.indexOf(node.root.health.status) > -1))
-        );
+            (filterInput.health.length === 0 || node.root.hook || (node.root.health && filterInput.health.indexOf(node.root.health.status) > -1)) &&
+            (filterInput.namespace.length === 0 || filterInput.namespace.includes(node.namespace)) &&
+            (filterInput.createdWithin.length === 0 || !!filterInput.createdWithin.find(v => createdWithin(v))) &&
+            (filterInput.label.length === 0 || !!filterInput.label.find(v => labels.includes(v)))
+        ) {
+            return true;
+        }
+
+        if (filterInput.ownership.includes('Owns') && ownership !== 'Owners') {
```

Comment:
reviewer - this is a recursive invocation - could have potential for large CPU cost with large app